hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/util/EntityAvroConverter.java

```diff
@@ -47,42 +52,28 @@ public static Entity convertToAvroEntity(@Nullable org.hypertrace.entity.data.se
-          case STRING:
-            result = AttributeValue.newBuilder().setValue(value.getValue().getString()).build();
-            break;
-          case INT:
-            result = AttributeValue.newBuilder().setValue(String.valueOf(value.getValue().getInt())).build();
-            break;
-          case LONG:
-            result = AttributeValue.newBuilder().setValue(
-                String.valueOf(value.getValue().getLong())).build();
-            break;
-          case BYTES:
-            result = AttributeValue.newBuilder().setBinaryValue(
-                value.getValue().getBytes().asReadOnlyByteBuffer()).build();
-            break;
-          case DOUBLE:
-            result = AttributeValue.newBuilder().setValue(
-                String.valueOf(value.getValue().getDouble())).build();
-            break;
-          case FLOAT:
-            result = AttributeValue.newBuilder().setValue(
-                String.valueOf(value.getValue().getFloat())).build();
-            break;
-          case TIMESTAMP:
-            result = AttributeValue.newBuilder().setValue(
-                String.valueOf(value.getValue().getTimestamp())).build();
-            break;
-          case BOOLEAN:
-            result = AttributeValue.newBuilder().setValue(
-                String.valueOf(value.getValue().getBoolean())).build();
-            break;
-          default:
-            LOG.warn("Unhandled entity attribute type: " + value.getValue().getTypeCase());
+        if (value.getValue().getTypeCase() == TypeCase.BYTES) {
+          result = AttributeValue.newBuilder().setBinaryValue(value.getValue().getBytes().asReadOnlyByteBuffer()).build();
+        } else {
+          Optional<String> valueStr = convertValueToString(value.getValue());
+          if (valueStr.isPresent()) {
+            result = AttributeValue.newBuilder().setValue(valueStr.get()).build();
+          }
         }
+      } else if (value.getTypeCase() == org.hypertrace.entity.data.service.v1.AttributeValue.TypeCase.VALUE_LIST) {
+        org.hypertrace.entity.data.service.v1.AttributeValueList attributeValueList = value.getValueList();
+        List<String> avroValuesStringList = new ArrayList<>();
+        for (org.hypertrace.entity.data.service.v1.AttributeValue attributeValue : attributeValueList.getValuesList()) {
+          if (attributeValue.getTypeCase() == org.hypertrace.entity.data.service.v1.AttributeValue.TypeCase.VALUE) {
+            Optional<String> valueStr = convertValueToString(attributeValue.getValue());
+            valueStr.ifPresent(avroValuesStringList::add);
+          } else {
+            LOG.warn("Unsupported entity attribute type of lists of lists or list of maps: {}", attributeValue);
+          }
+        }
+        result = AttributeValue.newBuilder().setValueList(avroValuesStringList).build();
       } else {
-        // Currently we don't copy the list or map types.
+        // Currently we don't copy the map types.
```

Comment:
Should we rate limit this log? Otherwise, this could flood the logs.

Reply:
err, i would not be worried about this one as much. I mean we just hit list type attributes and we are yet to have attributes that are maps in entities.